bw_check_commit_hook/data/checkpatch.yaml

```diff
@@ -0,0 +1,45 @@
+ERRORS_COMMON:
+  - TRAILING_WHITESPACE
+  - COMPLEX_MACRO
+  - DOS_LINE_ENDINGS
+  - DO_WHILE_MACRO_WITH_TRAILING_SEMICOLON
+  - EXECUTE_PERMISSIONS
+  - INVALID_UTF8
+  - LINE_CONTINUATIONS
+  - MACRO_ARG_PRECEDENCE
+  - MASK_THEN_SHIFT
+  - MISSING_EOF_NEWLINE
+  - MULTISTATEMENT_MACRO_USE_DO_WHILE
+  - ONE_SEMICOLON
+  - BRACKET_SPACE
+  - TYPO_SPELLING
+
+IGNORES_COMMON:
+  - BAD_SIGN_OFF
+  - SPDX_LICENSE_TAG
+  - FILE_PATH_CHANGES
+  - NOT_UNIFIED_DIFF
+  - LINUX_VERSION_CODE
+  - CONSTANT_COMPARISON
+  - OPEN_ENDED_LINE
+  - UNNECESSARY_PARENTHESES
+  - GERRIT_CHANGE_ID
+  - COMMIT_LOG_LONG_LINE
+  - EMAIL_SUBJECT
+  - GIT_COMMIT_ID
+
+# Per-directory checkpatch configuration
+DIR_CONFIGS:
+  .:
```

Comment:
This syntax doesn't seem ideal - could it at least become `"."` to make it a little clearer?

Reply:
I can add a magic string `all` and then handle the conversion in the code, WDYT?

Reply:
`all` makes it sound like it would apply in all sub-directories regardless of whether they have their own entries in the YAML. Maybe `default` would be better? Or maybe there's a better convention out there somewhere.

Reply:
`default` works for me.